homeassistant/components/bluetooth_tracker/device_tracker.py

```diff
@@ -56,22 +58,19 @@ def is_bluetooth_device(device: Device) -> bool:
     return device.mac is not None and device.mac[:3].upper() == BT_PREFIX
 
 
-def discover_devices(device_id: int) -> list[tuple[str, str]]:
+async def discover_devices(hass: HomeAssistant) -> list[tuple[str, str]]:
     """Discover Bluetooth devices."""
-    try:
-        result = bluetooth.discover_devices(
-            duration=8,
-            lookup_names=True,
-            flush_cache=True,
-            lookup_class=False,
-            device_id=device_id,
-        )
-    except OSError as ex:
-        # OSError is generally thrown if a bluetooth device isn't found
-        _LOGGER.error("Couldn't discover bluetooth devices: %s", ex)
-        return []
-    _LOGGER.debug("Bluetooth devices discovered = %d", len(result))
-    return result  # type: ignore[no-any-return]
+    devices = await bluetooth.async_get_scanner(hass).discover()
```

Comment:
All the discovered devices can be fetched with https://developers.home-assistant.io/docs/core/bluetooth/api?_highlight=bluetoot#fetching-all-discovered-devices